core-bundle/src/Resources/contao/library/Contao/Model.php

```diff
@@ -750,9 +738,7 @@ public static function findByPk($varValue, array $arrOptions=array())
 	{
 		if ($varValue === null)
 		{
-			trigger_deprecation('contao/core-bundle', '4.13', 'Passing "null" as primary key has been deprecated and will no longer work in Contao 5.0.', __CLASS__);
-
-			return null;
+			throw new \TypeError('Model::findByPk(): Argument #1 ($varValue) must not be of type null.');
```

Comment:
shouldn't we just add `string|int` as type argument instead?

Reply:
Thought about this as well, but this is IMHO better be done in a separate PR because it also affects all child classes. Maybe we should decide which of the legacy classes should receive type hints and then try to automate those changes as much as possible with tooling.

Reply:
In my StringUtil PR we added only the type hints required and left out all the other stuff which can still be improved with tooling :)